scripts/irods/upgrade_configuration.py

```diff
@@ -355,11 +355,13 @@ def update_base(base, updates):
         database_config_options = new_server_config['plugin_configuration']['database']
 
         # Assume we have a previously valid config (i.e. one database config property)
-        # MAYBE: If there are other configurations, we should delete them
-        # or just do something (e.g. print a warning?)
         db_configurations = list(database_config_options.keys())
         db_name = db_configurations.pop(0)
 
+        # Check if there are multiple database plugin configurations
+        if len(db_configurations) != 0:
+            l.warning('Extra database plugin configurations discovered. Server startup will likely fail.')
```

Comment:
Could we change "Server startup will likely fail." to "Schema validation will fail. Check server configuration."? A little more firm and precise, plus an action step. Thoughts?

Reply:
Agreed

Reply:
Does the server support custom schemas? I thought it did, so I put 'likely fail' just in case. We could cut the last sentence and just put `Check server configuration`.

I'm open to the full `Schema validation will fail. Check server configuration` suggestion too.

Reply:
> Does the server support custom schemas?

That'd be news to me. If we *support* custom schemas for server configuration, then shouldn't the server function as normal in this case? Hmm...

Reply:
We used to.  But since 4.2…. We ship them.

Reply:
I don't think we need to worry about custom schemas. If someone chooses to modify the schema, that's on them.

The only thing we can assert about the situation is that additional unexpected information was found.

Do we know for sure if schema validation will fail if additional properties are included?

Reply:
That is up to the schema.  We generally don’t set that.

Reply:
Okay. If we don't know whether unexpected config properties will cause validation to fail, we should avoid stating that. That means the message should cover the following:
- Report that unexpected properties were found
- Instruct the admin to check the database configuration stanza for correctness

Reply:
That works, but I think the first statement can be improved. Here's my suggestion.
```
Database plugin configuration contains unexpected properties. Check server configuration.
```
or ...
```
Unexpected properties found in database plugin configuration. server_config.json needs attention.
```

It doesn't have to be that exact wording, but including words like "unexpected" should catch the admin's attention even more.

Reply:
Done.